ddtrace/appsec/_iast/__init__.py

```diff
@@ -69,8 +70,12 @@ def ddtrace_iast_flask_patch():
         return
 
     compiled_code = compile(patched_ast, module_path, "exec")
+    # creating a new module environment to execute the patched code from scratch
+    new_module = types.ModuleType(module_name)
+    module.__dict__.clear()
+    module.__dict__.update(new_module.__dict__)
+    # executing the compiled code in the new module environment
```

Comment:
😍

Reply:
This is a little magic, but the cleanest way I found to reinitialise properly the module without unloading it (as we can't here, because it's __main__).